AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/Statement+Model.swift

```diff
@@ -35,99 +33,159 @@ protocol StatementModelConvertible {
+            guard let (schema, field) = columnMapping[column] else {
                 logger.debug("""
-                A column named \(name) was found in the result set but no field on
-                \(schema.name) could be found with that name and it will be ignored.
+                A column named \(column) was found in the result set but no field on
+                \(modelType.modelName) could be found with that name and it will be ignored.
                 """)
+                continue
+            }
+
+            let modelValue = try SQLiteModelValueConverter.convertToSource(
+                from: value,
+                fieldType: field.type
+            )
+            modelDictionary.updateValue(modelValue, forKeyPath: column)
+
+            // create lazy list for "many" relationships
+            if let id = modelValue as? String, field.isPrimaryKey {
+                let associations = schema.fields.values.filter {
+                    $0.isArray && $0.hasAssociation
+                }
+                let prefix = field.name.replacingOccurrences(of: field.name, with: "")
+                associations.forEach { association in
+                    let associatedField = association.associatedField?.name
+                    let lazyList = List<AnyModel>.lazyInit(associatedId: id,
+                                                           associatedWith: associatedField)
+                    let listKeyPath = prefix + association.name
+                    modelDictionary.updateValue(lazyList, forKeyPath: listKeyPath)
+                }
             }
         }
+        // swiftlint:disable:next force_cast
+        return modelDictionary as! ModelValues
+    }
 
-        if let id = values[schema.primaryKey.name] as? String {
-            // create instances of lazy-load lists of fields that represent "many" associations
-            schema.fields.values.filter { $0.isArray }.forEach { field in
-                // TODO extract this to a List utility
-                let lazyListDecodable: [String: Any?] = [
-                    "associatedId": id,
-                    "associatedField": field.associatedField?.name,
-                    "elements": []
-                ]
-                values[field.name] = lazyListDecodable
+}
+
+internal extension List {
+
+    /// Creates a data structure that is capable of initializing a `List<M>` with
+    /// lazy-load capabilities when the list is being decoded.
+    ///
+    /// See the `List.init(from:Decoder)` for details.
+    static func lazyInit(associatedId: String, associatedWith: String?) -> [String: Any?] {
+        return [
+            "associatedId": associatedId,
+            "associatedField": associatedWith,
+            "elements": []
+        ]
+    }
+}
+
+internal extension Dictionary where Key == String, Value == Any? {
+
+    /// Utility to create a `NSMutableDictionary` from a Swift `Dictionary<String, Any?>`.
+    func mutableCopy() -> NSMutableDictionary {
```

Comment:
I think this name should be something like nsMutableCopy... because we are converting a data structure (dictionary) to an NS data structure?

Reply:
sure... but doesn't the return type make that clear?